src/main/java/com/limechain/utils/json/JsonParser.java

```diff
@@ -142,7 +142,7 @@ private String parseString() {
 
     private Number parseNumber() {
         int start = index;
-        while (Character.isDigit(peek()) || peek() == '-' || peek() == '.' || peek() == 'e' || peek() == 'E') {
+        while (index < json.length() && (Character.isDigit(json.charAt(index)) || json.charAt(index) == '-' || json.charAt(index) == '.' || json.charAt(index) == 'e' || json.charAt(index) == 'E')) {
```

Comment:
This line seems way too long and hard to read. Can you take part of the logic and assign it to boolean variables so that the `while` line is short and easy to read?